measure_vram.py

```diff
@@ -0,0 +1,205 @@
+                  f"and {num_iterations_for_vram} iterations. Using all available samples.")
+            base_ds_for_vram_test = base_ds_full
+        else:
+            base_ds_for_vram_test = base_ds_full.select(range(required_samples_for_base_ds))
+        print(f"Using {len(base_ds_for_vram_test)} samples for VRAM testing.")
+    except Exception as e:
+        print(f"Error loading dataset: {dataset_path}, name: {dataset_name}. Error: {e}")
+        print("Please ensure the dataset path and name are correct.")
+        return
+
+    processed_base_dataset = VQADataset(base_ds_for_vram_test, tokenizer, image_processor)
+    vqa_collator = VQACollator(tokenizer, vlm_cfg.lm_max_length)
+
+    print("\n--- VRAM Measurement ---")
+    results = {}
+
+    for bs in batch_sizes_to_test:
+        print(f"\nTesting Batch Size: {bs}")
+        
+        if len(processed_base_dataset) < bs:
+            print(f"Base processed dataset has {len(processed_base_dataset)} samples, "
+                  f"not enough for batch size {bs}. Skipping.")
+            results[bs] = "Not enough data"
+            continue
+
+        current_loader = DataLoader(
+            processed_base_dataset,
+            batch_size=bs,
+            shuffle=False, 
+            collate_fn=vqa_collator,
+            num_workers=args.num_workers,
+            pin_memory=True,
+            drop_last=True # Important if dataset size is not exactly multiple of bs
+        )
+
+        if len(current_loader) < num_iterations_for_vram:
+             print(f"Dataloader for batch size {bs} yields {len(current_loader)} batches, "
+                   f"less than requested {num_iterations_for_vram} iterations. Will run available batches.")
+             if len(current_loader) == 0:
+                 print(f"Dataloader for batch size {bs} is empty. Skipping.")
+                 results[bs] = "Dataloader empty"
+                 continue
+
+
+        # Reset CUDA memory stats for each batch size test
+        torch.cuda.reset_peak_memory_stats(device)
+        
+        # Model to train mode for realistic scenario (e.g. dropout layers active)
+        model.train() 
+        optimizer = optim.AdamW(model.parameters(), lr=1e-5) # Dummy optimizer
+
+        try:
+            for i, batch in enumerate(current_loader):
+                if i >= num_iterations_for_vram:
+                    break
+                
+                images = batch["image"].to(device)
+                input_ids = batch["input_ids"].to(device)
+                labels = batch["labels"].to(device)
+                attention_mask = batch["attention_mask"].to(device)
+
+                optimizer.zero_grad(set_to_none=True)
+
+                with torch.autocast(device_type='cuda', dtype=torch.bfloat16 if torch.cuda.is_bf16_supported() else torch.float16):
+                    _, loss = model(input_ids, images, attention_mask=attention_mask, targets=labels)
```

Comment:
I am sceptical about this. If we do autocast, we should be printing that out as information. WDYT?